### PR TITLE
TMDM-13276 [Impact Analysis] ElementType change (length) not reflected in PostgreSQL database tables

### PR DESCRIPTION
**Pull request by zfkang-talend**
Jira: https://jira.talendforge.org/browse/TMDM-13276

**What is the current behavior?** (You should also link to an open issue here)
1. Enlarge the String field length, greater than 255, H2, MySQL, PostgreSQL, DB2, after redeploy, column type doesn't modify in the database.

**What is the new behavior?**
1. if enlarge the string field length greater than 255, H2, MySQL, PostgreSQL, after redeploy, column type changed to text from varchar.
2. DB2: DB2 doesn't support changed the type from varchar to clob.
from: https://www.ibm.com/support/knowledgecenter/en/SSEPGG_11.1.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000888.html
> A non-LOB column cannot be altered to a LOB data type (SQLSTATE 42837).
So, for DB2, don't deal with this case, maintain the same with before.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java [hunk 1]
```diff
@@ -290,6 +281,17 @@ public Iterator sqlAlterStrings(Dialect dialect, Mapping p, TableMetadata tableI
         return results.iterator();
     }
 
+    private String generateUK(Dialect dialect, Column col) {
+        StringBuilder buf = new StringBuilder();
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java]: Don't think we need to use StringBuilder here, just need to return `dialect.getUniqueDelegate().getColumnDefinitionUniquenessFragment(col)`

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java [hunk 2]
```diff
@@ -290,6 +281,16 @@ public Iterator sqlAlterStrings(Dialect dialect, Mapping p, TableMetadata tableI
         return results.iterator();
     }
 
+    private String generateUK(Dialect dialect, Column col) {
+        if (col.isUnique()) {
+            String keyName = Constraint.generateName("UK_", this, col);
```
Comment (hw-zhu) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java]: //$NON-NLS-1$

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTableUtils.java [hunk 3]
```diff
@@ -24,29 +27,34 @@ public static boolean isAlterColumnField(Column newColumn, ColumnMetadata oldCol
...
         }
-        boolean isNeedToAlter = Boolean.TRUE;
-        isNeedToAlter &= isVarcharField(newColumn, oldColumnInfo, dialect)
-                && isIncreaseVarcharColumnLength(newColumn, oldColumnInfo, dialect);
-        return isNeedToAlter;
+        return isVarcharField(oldColumnInfo, dialect) && isIncreaseVarcharColumnLength(newColumn, oldColumnInfo)
+                || isStringTypeChanged(newColumn, oldColumnInfo, dialect);
+    }
+
+    public static boolean isStringTypeChanged(Column newColumn, ColumnMetadata oldColumnInfo, Dialect dialect) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTableUtils.java]: Can the method name be `isVarcharTypeChanged`?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java (hunk 2 above)
Comment (zfkang-talend) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java]: I had comment before class:
`@SuppressWarnings({ "nls", "rawtypes", "deprecation", "serial", "unchecked" })`

**Pull request merged**